lib/top_score_filter.py

```diff
@@ -69,19 +70,22 @@ def execute(self):
                     if len(lines) == 1:
                         filtered_lines.append(lines[0])
                     else:
-                        lines_with_transcript_expression = list(filter(lambda line: line['Transcript Expression'] != 'NA' in lines))
+                        lines_with_transcript_expression = list(filter(lambda line: line['Transcript Expression'] != 'NA', lines))
                         if len(lines_with_transcript_expression) > 0:
                             line_with_max_expression = lines_with_transcript_expression[0]
                             for line_with_transcript_expression in lines_with_transcript_expression:
                                 if float(line_with_transcript_expression['Transcript Expression']) > float(line_with_max_expression['Transcript Expression']):
                                     line_with_max_expression = line_with_transcript_expression
                             filtered_lines.append(line_with_max_expression)
                         else:
-                            line_with_lowest_transcript_no = lines[0]
+                            line_with_lowest_transcript_id = lines[0]
+                            lowest_transcript_id = re.compile('ENST(\d+)').match(line_with_lowest_transcript_id['Transcript']).group(1)
                             for line in lines:
-                                if float(line['Transcript'].replace('ENST','')) < float(line_with_lowest_transcript_no['Transcript'].replace('ENST','')):
-                                    line_with_lowest_transcript_no = line
-                            filtered_lines.append(line_with_lowest_transcript_no)
+                                transcript_id = re.compile('ENST(\d+)').match(line['Transcript']).group(1)
+                                if transcript_id < lowest_transcript_id:
+                                    lowest_transcript_id = re.compile('ENST(\d+)').match(line['Transcript']).group(1)
```

Comment:
Seems like this could be:
```suggestion
                                    lowest_transcript_id = transcript_id
```

Reply:
Doh, of course.